src/compiler/crystal/tools/formatter.cr

```diff
@@ -1555,7 +1559,7 @@ module Crystal
       end
 
       args.each_with_index do |arg, i|
-        has_more = !last?(i, args) || double_splat || block_arg || yields || variadic
+        has_more = !last?(i, args) || double_splat || block_arg || (yields && flag?("method_signature_yield")) || variadic
```

Comment:
This overload can actually be left intact by modifying the other one:

```crystal
    def format_def_args(node : Def | Macro)
      yields = node.is_a?(Def) && !node.block_arity.nil? && flag?("method_signature_yield")
      format_def_args node.args, node.block_arg, node.splat_index, false, node.double_splat, yields
    end
```

Reply:
This overload is semantically equivalent to reverting #12951 and #13126 if `yields` is replaced with `yields && flag?("method_signature_yield")` everywhere when the node is a `Def`. This current revision isn't and actually misses the double splat case, formatting:

```crystal
def foo(a, **b)
  yield
end
```

to:

```crystal
def foo(a, **b,)
  yield
end
```

(To be fair I totally forgot about adding that one as a spec)

If `yields` sounds wrong perhaps it should be renamed to something else. Already it doesn't universally refer to yielding entities because it is false when called from a block-accepting macro.